platforms/azure/giantnetes/main.tf

```diff
@@ -81,14 +100,89 @@ module "vault" {
   network_interface_ids   = "${module.vnet.vault_network_interface_ids}"
   resource_group_name     = "${module.resource_group.name}"
   storage_type            = "${var.vault_storage_type}"
+  user_data               = "${data.ct_config.vault.rendered}"
   vm_size                 = "${var.vault_vm_size}"
 }
 
+# Generate ignition config for master.
+data "template_file" "master_big" {
+  template = "${file("${path.module}/../../../ignition/azure/master.yaml.tmpl")}"
+
+  vars {
+    "API_DOMAIN_NAME"          = "${var.api_dns}.${var.base_domain}"
+    "AZURE_CLOUD"              = "${var.azure_cloud}"
+    "AZURE_LOCATION"           = "${var.azure_location}"
+    "AZURE_SP_TENANTID"        = "${var.azure_sp_tenantid}"
+    "AZURE_SP_SUBSCRIPTIONID"  = "${var.azure_sp_subscriptionid}"
+    "AZURE_SP_AADCLIENTID"     = "${var.azure_sp_aadclientid}"
+    "AZURE_SP_AADCLIENTSECRET" = "${var.azure_sp_aadclientsecret}"
+    "AZURE_RESOURCEGROUP"      = "${var.cluster_name}"
+    "AZURE_SUBNETNAME"         = "${var.cluster_name}_worker_subnet"
+    "AZURE_SECGROUPNAME"       = "${var.cluster_name}-worker"
+    "AZURE_VNETNAME"           = "${var.cluster_name}"
+    "AZURE_ROUTETABLE"         = "${var.cluster_name}_worker_rt"
+    "POD_CIDR"                 = "${var.pod_cidr}"
+    "K8S_SERVICE_CIDR"         = "${var.k8s_service_cidr}"
+    "K8S_API_IP"               = "${var.k8s_api_ip}"
+    "CLUSTER_NAME"             = "${var.cluster_name}"
+    "DEFAULT_IPV4"             = "$${DEFAULT_IPV4}"
+    "DOCKER_CIDR"              = "${var.docker_cidr}"
+    "ETCD_DOMAIN_NAME"         = "${var.etcd_dns}.${var.base_domain}"
+    "G8S_VAULT_TOKEN"          = "${var.nodes_vault_token}"
+    "K8S_DNS_IP"               = "${var.k8s_dns_ip}"
+    "VAULT_DOMAIN_NAME"        = "${var.vault_dns}.${var.base_domain}"
+  }
+}
+
+# Convert ignition config to raw json and merge users part.
+data "ct_config" "master_big" {
+  content      = "${format("%s\n%s", local.ignition_users, data.template_file.master_big.rendered)}"
+  platform     = "azure"
+  pretty_print = false
+}
+
+resource "local_file" "master_ignition_big" {
+  content  = "${data.ct_config.master_big.rendered}"
+  filename = "${path.cwd}/generated/master-ignition.yaml"
+}
+
+resource "azurerm_storage_account" "storage_acc" {
+  name                     = "${var.cluster_name}machine"
+  resource_group_name      = "${module.resource_group.name}"
+  location                 = "westeurope"
+  account_tier             = "Standard"
```

Comment:
also `account_kind=BlobStorage` and `access_tier=Cold` , because we need only objects and not going to host VM disks there.